src/data/markdown/translated-guides/en/02 Using k6/05 k6 Options/01 How to.md

```diff
@@ -0,0 +1,166 @@
+
+*Options in the script or config file let you version control your options (and keep your tests tidy).*
+
+You may wonder why you would need a config file when you can use the script options.
+Isn't this redundant?
+In fact, you might have multiple reasons to set your options in a config file.
+
+For one, k6 uses the (auto-generated) config file to keep users logged in after they run `k6 login cloud`.
+Other reasons for config files are more about developer style and comfort:
+- To separate the test configuration from JavaScript code
+- To reduce the size of the entry-point script, especially when there are multiple scenarios
+- To more easily switch between configs (either rename the current config file or modify the JavaScript code in one location)
+
+
+## Examples of setting options
+
+The following JS snippets show some examples of how you can set options.
+
+### Options in the script
+
+<CodeGroup labels={["example.js"]} lineNumbers={[true]}>
+
+```javascript
+import http from 'k6/http';
+
+export const options = {
+  hosts: { 'test.k6.io': '1.2.3.4' },
+  stages: [
+    { duration: '1m', target: 10 },
+    { duration: '1m', target: 20 },
+    { duration: '1m', target: 0 },
+  ],
+  thresholds: { http_req_duration: ['avg<100', 'p(95)<200'] },
+  noConnectionReuse: true,
+  userAgent: 'MyK6UserAgentString/1.0',
+};
+
+export default function () {
+  http.get('http://test.k6.io/');
+}
+```
+
+</CodeGroup>
+
+### Options in a config file
+
+<div id="config-json-example">
+You can also define the same options through a config file:
+</div>
+
+<CodeGroup labels={["config.json"]} lineNumbers={[true]}>
+
+```json
+{
+  "hosts": {
+    "test.k6.io": "1.2.3.4"
+  },
+  "stages": [
+    {
+      "duration": "1m",
+      "target": 10
+    },
+    {
+      "duration": "1m",
+      "target": 30
+    },
+    {
+      "duration": "1m",
+      "target": 0
+    }
+  ],
+  "thresholds": {
+    "http_req_duration": ["avg<100", "p(95)<200"]
+  },
+  "noConnectionReuse": true,
+  "userAgent": "MyK6UserAgentString/1.0"
+}
+```
+
+</CodeGroup>
+
+### Options with environment variables
+
+Set some of the previous options via environment variables and command-line flags:
+
+<CodeGroup labels={["Bash", "Windows: CMD", "Windows: PowerShell"]} lineNumbers={[false]}>
+
+```bash
+$ K6_NO_CONNECTION_REUSE=true K6_USER_AGENT="MyK6UserAgentString/1.0" k6 run script.js
+
+$ k6 run --no-connection-reuse --user-agent "MyK6UserAgentString/1.0" script.js
+```
+
+```bash
+C:\k6> set "K6_NO_CONNECTION_REUSE=true" && set "K6_USER_AGENT=MyK6UserAgentString/1.0" && k6 run script.js
+
+C:\k6> k6 run --no-connection-reuse --user-agent "MyK6UserAgentString/1.0" script.js
+```
+
+```bash
+PS C:\k6> $env:K6_NO_CONNECTION_REUSE=true; $env:K6_USER_AGENT="MyK6UserAgentString/1.0"; k6 run script.js
+
+PS C:\k6> k6 run --no-connection-reuse --user-agent "MyK6UserAgentString/1.0" script.js
+```
+
+</CodeGroup>
+
```

Comment:
Another popular way of getting options is to set them on CLI as k6 variables. (not the same as env variables).

```
k6 run script.js -e MY_USER_AGENT="hello"
```

and then in the script: 

```
import http from 'k6/http';

export const options = {
    userAgent: __ENV.MY_USER_AGENT
};

export default function () {
  http.get('http://test.k6.io/');
}

```

This is very flexible as it allows users to add more logic in how they configure the execution